python-squid/src/main/java/org/sonar/python/tree/TreeUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.tree;
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+import javax.annotation.CheckForNull;
+import org.sonar.python.api.tree.Tree;
+import org.sonar.python.api.tree.Tree.Kind;
+
+public class TreeUtils {
+  private TreeUtils() {
+    // empty constructor
+  }
+
+  @CheckForNull
+  public static Tree firstAncestor(Tree tree, Predicate<Tree> predicate) {
+    Tree currentParent = tree.parent();
+    while (currentParent != null) {
+      if (predicate.test(currentParent)) {
+        return currentParent;
+      }
+      currentParent = currentParent.parent();
+    }
+    return null;
+  }
+
+  @CheckForNull
+  public static Tree firstAncestorOfKind(Tree tree, Kind... kinds) {
+    return firstAncestor(tree, t -> Arrays.asList(kinds).contains(t.getKind()));
```

Comment:
What about changing Tree#is  to be variadic ? 
`is(Kind... kinds)` and thus simplify this call ?